lib/puppet/rest/response.rb

```diff
@@ -0,0 +1,26 @@
+module Puppet::Rest
+  # This is a wrapper for the HTTP::Message class of the HTTPCLient
+  # gem. It is designed to wrap a message sent as an HTTP response.
+  class Response
+
+    def initialize(message)
+      @message = message
+    end
+
+    def body
+      @message.body
```

Comment:
Puppet's legacy rest code added methods to stream request/response bodies after the fact, and those methods behave inconsistently, e.g. retries aren't implemented consistently if you pass a block vs don't. Could we add streaming to the API to start with so we can't accidentally read large files into memory?

Reply:
Would we want to always stream?

Reply:
I think always streaming now will be easier in the long run as it will have an impact on the `get` APIs. But just a suggestion, not blocking this PR.

Reply:
Filed https://tickets.puppetlabs.com/browse/PUP-8702 about the streaming. I've taken a look at the API and it shouldn't be too hard to add.